app/models/mixins/miq_web_server_worker_mixin.rb

```diff
@@ -87,8 +87,7 @@ def self.sync_workers
             _log.info("Reserved port=#{port}, Current ports in use: #{ports.inspect}")
             ports << port
             ports_hash[:adds] << port
-            w = start_worker(:uri => build_uri(port))
-            result[:adds] << w.pid
+            result[:adds] << Proc.new { start_worker(:uri => build_uri(port)) }
```

Comment:
Consider introducing an object (`result.launch { start_worker(:uri => build_uri(port)) }`) instead of directly exposing so many primitives? (hash of symbols to arrays of procs :neutral_face:)